mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,874 @@
+        local_model_path=local_sparkml_model_path,
+    )
+
+
+def _load_pyfunc(path, spark=None):
+    """
+    Load PyFunc implementation. Called by ``pyfunc.load_model``.
+    :param path: Local filesystem path to the MLflow Model with the ``johnsnowlabs`` flavor.
+    :param spark: Optionally pass spark context when using pyfunc as UDF. required, because we cannot fetch the Sparkcontext inside of the Workernode which executes the UDF
+    :return:
+    """
+    return _PyFuncModelWrapper(_load_model(model_uri=path), spark if spark else get_or_create_sparksession(path))
+
+
+def get_or_create_sparksession(model_path=None):
+    """
+    1. Check if SparkSession running and get it
+    2. If none exists, create a new one using jars in model_path
+    3. If model_path not defined, rely on nlp.start() to create a new
+    one using johnsnowlabs Jar resolution method
+    See https://nlp.johnsnowlabs.com/docs/en/jsl/start-a-sparksession
+    and https://nlp.johnsnowlabs.com/docs/en/jsl/install_advanced
+    :param model_path:
+    :return:
+    """
+    validate_env_vars()
+    from mlflow.utils._spark_utils import _get_active_spark_session
+    spark = _get_active_spark_session()
+    if spark is None:
+        spark_conf = {}
+        spark_conf["spark.python.worker.reuse"] = "true"
+        # disable task retry (i.e. make it fast fail)
+        spark_conf["spark.task.maxFailures"] = "1"
+        # Disable simplifying traceback from Python UDFs
+        spark_conf["spark.sql.execution.pyspark.udf.simplifiedTraceback.enabled"] = "false"
+        # Show jvm side stack trace.
+        spark_conf["spark.sql.pyspark.jvmStacktrace.enabled"] = "true"
+        spark_conf["spark.jars.excludes"] = "net.sourceforge.f2j:arpack_combined_all"
+        spark_conf["spark.driver.extraJavaOptions"] = "-Dio.netty.tryReflectionSetAccessible=true"
+        spark_conf["spark.executor.extraJavaOptions"] = "-Dio.netty.tryReflectionSetAccessible=true"
+
+        os.environ['PYSPARK_PYTHON'] = sys.executable
+        os.environ['PYSPARK_DRIVER_PYTHON'] = sys.executable
+        if model_path:
+            jar_paths, license_path = fetch_deps_from_path(os.path.join(model_path))
+            # jar_paths += get_mleap_jars().split(',')  # TODO when to load MLleap Jars
+            if license_path:
+                if 'JSL_NLP_LICENSE' not in os.environ:
+                    os.environ.update({k: str(v) for k, v in json.load(open(license_path)).items() if v is not None})
+                    os.environ['JSL_NLP_LICENSE'] = json.load(open(license_path))['HC_LICENSE']
+            print("STARTING NEW  SESSION WITH JARS: ", jar_paths)
+            spark = nlp.start(nlp=False, spark_nlp=False, jar_paths=jar_paths, json_license_path=license_path,
+                              create_jsl_home_if_missing=False, spark_conf=spark_conf)
+        else:
+            spark = nlp.start()
+    return spark
+
+
+def create_sparksession_for_jsl_uri(**kwargs):
```

Comment:
Where is this function used? Is this part of the flavor in any way?
If it's a utility to make the session initialize, this should just be used as an end-user would normally do this via the API call that this function wraps.

Reply:
Sorry, this function is irrelevant, removed it thanks for the hint.